Sources/MixedReality/Microsoft.Psi.MixedReality.UniversalWindows/MediaCaptureMicrophone.cs

```diff
@@ -0,0 +1,253 @@
+            {
+                var frameSource = this.mediaCapture.FrameSources[sourceInfo.Value.Info.Id];
+                return await this.mediaCapture.CreateFrameReaderAsync(frameSource);
+            }
+
+            // No frame source was found for the requested stream type.
+            return null;
+        }
+
+        /// <summary>
+        /// Creates an event handler that handles the FrameArrived event of the MediaFrameReader.
+        /// </summary>
+        /// <param name="audioSettings">The stream settings.</param>
+        /// <param name="audioStream">The stream on which to post the audio buffer.</param>
+        /// <returns>The event handler.</returns>
+        private TypedEventHandler<MediaFrameReader, MediaFrameArrivedEventArgs> CreateMediaFrameHandler(
+            MediaCaptureMicrophoneConfiguration audioSettings,
+            Emitter<AudioBuffer> audioStream)
+        {
+            return (sender, args) =>
+            {
+                using var frame = sender.TryAcquireLatestFrame();
+                if (frame != null)
+                {
+                    // Convert frame QPC time to pipeline time
+                    var frameTimestamp = frame.SystemRelativeTime.Value.Ticks;
+                    var originatingTime = this.pipeline.GetCurrentTimeFromElapsedTicks(frameTimestamp);
+
+                    // Post the audio buffer stream if requested
+                    if (audioSettings.OutputAudio)
+                    {
+                        using MediaFrameReference mediaFrame = frame.AudioMediaFrame.FrameReference;
+                        using AudioFrame audioFrame = frame.AudioMediaFrame.GetAudioFrame();
+
+                        AudioEncodingProperties audioEncodingProperties = mediaFrame.AudioMediaFrame.AudioEncodingProperties;
+
+                        unsafe
+                        {
+                            using Windows.Media.AudioBuffer buffer = audioFrame.LockBuffer(AudioBufferAccessMode.Read);
+                            using IMemoryBufferReference reference = buffer.CreateReference();
+                            ((UnsafeNative.IMemoryBufferByteAccess)reference).GetBuffer(out byte* audioDataIn, out uint capacity);
+
+                            uint frameDurMs = (uint)mediaFrame.Duration.TotalMilliseconds;
+                            uint sampleRate = audioEncodingProperties.SampleRate;
+                            uint sampleCount = (frameDurMs * sampleRate) / 1000;
+
+                            uint bytesPerSample = audioEncodingProperties.BitsPerSample / 8;
+                            uint numAudioChannels = audioEncodingProperties.ChannelCount;
+
+                            // Buffer size is (number of samples) * (size of each sample)
+                            byte[] audioDataOut = new byte[sampleCount * bytesPerSample];
+
+                            // Convert to bytes
+                            if (numAudioChannels > this.audioFormat.Channels)
```

Comment:
Per discussion in the PR, instead of relying on `AudioFormat.Channels` in the configuration, you could instead have a flag in your configuration indicating whether or not to downsample or output raw audio, e.g.
```
if (this.configuration.SingleChannel && numAudioChannels > 1)
```